sdk/core/azure_core_amqp/src/fe2o3/messaging/message_source.rs

```diff
@@ -202,6 +196,8 @@ fn message_source_conversion_fe2o3_amqp() {
             original_default_outcome.is_modified(),
             round_trip_default_outcome.is_modified(),
         );
+    } else {
+        panic!("Expected default outcomes");
```

Comment:
The others look fine, but I really don't understand the issue here. At the very least, I'd expect a change to be more like:

```rust
if let Some(original_default_outcome) = fe2o3_source.default_come {
    let round_trip_default_outcome = round_trip.default_outcome.unwrap();
    // ...
}
```

But adding the panic if `fe2o3.source.default_outcome` is `None` changes the logic here in ways that will crash more often.

Reply:
~~Today the test asserts that both Options are Some, so unless I'm misunderstanding something (please clue me in) this change doesn't introduce a new panic condition~~

Edit: 🤣 saw it immediately after submitting this comment: test asserts that both are Some or both are None. I'll push a commit to restore that condition